### PR TITLE
Update documentation for 4.0

### PR DESCRIPTION
**Pull request by gerhardol**
Replacing #139

Plan to update the FormBrowse description some more, no more plan to change other pages now (there is a need to change more though...)

### CODE REVIEW

File: source/settings.rst [hunk 1]
```diff
@@ -447,677 +445,785 @@ depending on whether or not they are distributed with that repository.
...
+  This page allows you to define keyboard shortcuts to actions when specific pages of Git Extensions are displayed.
+  The HotKeyable Items identifies a page within Git Extensions. Selecting a Hotkeyable Item displays the list of
+  commands on that page that can have a hotkey associated with them.
+  The Hotkeyable Items consist of the following pages
 
-    This page allows you to configure specific commands to run before/after Git actions or to add a new command to the User Menu.
-    The top half of the page summarises all of the scripts currently defined. If a script is selected from the summary, the bottom
-    half of the page will allow modifications to the script definition.
+  #. Commit: the page displayed when a Commit is requested via the ``Commit`` User Menu button or the ``Commands/Commit`` menu option.
+  #. Browse: the Revision Graph page (the page displayed after a repository is selected from the dashboard (Start Page)).
```
Comment (mstv) [inline source/settings.rst:627]:
```suggestion
  #. Browse: the Revision Graph page (the page displayed after a repository is selected from the dashboard (Start Page)).
  #. LeftPanel: ...
```

File: source/settings.rst [hunk 2]
```diff
@@ -447,677 +445,785 @@ depending on whether or not they are distributed with that repository.
...
+.. settingsgroup:: Confirm actions
 
-      .. settingbutton:: Clear
+  .. settingsgroup:: Commits
 
-        Sets the keyboard shortcut for the currently selected Command to 'None'.
+  .. setting:: Amend last commit
 
-    .. settingbutton:: Reset all Hotkeys to defaults
+    If checked, do not display the popup warning about
```
Comment (mstv) [inline source/settings.rst:738]: These settings have been inverted.
```suggestion
    If not checked, do not display the popup warning about
```
or
```suggestion
    If checked, display the popup warning about
```

**Review by mstv — COMMENTED** [review]
👍

File: source/settings.rst (hunk 2 above)
Comment (gerhardol) [reply source/settings.rst]: ok

File: source/settings.rst (hunk 1 above)
Comment (gerhardol) [reply source/settings.rst]: ok

File: source/file_history.rst [hunk 3]
```diff
@@ -1,8 +1,15 @@
+.. _file-history:
+
 File history
 ============
 
+``File tree`` is a separate form to view the history of a file or folder.
```
[the remark this answers is not in the document]
Comment (gerhardol) [reply source/file_history.rst]: ok

File: source/install.rst [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+.. figure:: /images/install/options.png
+
+    Choose the options to install.
+
+.. image:: /images/install/ready.png
+
+Portable
+--------
+
+Git Extensions is also distributed as a portable .zip file, that only require unpacking to a new directory (migrate settings files and theme manually).
```
[the remark this answers is not in the document]
Comment (gerhardol) [reply source/install.rst]: ok

**Comment by gerhardol** [comment]
Updated build
While Browse may be improved a lot, it could be merged right now

File: source/settings.rst (hunk 2 above)
Comment (mstv) [reply source/settings.rst]: This applies to all confirmation settings.

File: source/browse_repository.rst [hunk 5]
```diff
@@ -35,26 +54,96 @@ commit with matching text. You can search for the next or previous commit with m
...
+
 .. image:: /images/advance_filter_dialog.png
+
+.. _browse-reflog:
+
+Reflog
+""""""
+
+By default Git will not show commits that has no branch or tag on a child. Such commits will be purged when Git runs maintenance.
+By enabling Reflog (toolbar button or ``View`` menu) these commits will be shown too.
```
Comment (RussKie) [inline source/browse_repository.rst:99]:
```suggestion
By default, Git will not show any commits that are not reachable and don't have any references, such as a branch or a tag. Such commits will be purged when Git runs maintenance.
By enabling Reflog via the toolbar button or in the ``View`` menu these commits will be shown too.
```

File: source/browse_repository.rst [hunk 6]
```diff
@@ -35,26 +54,96 @@ commit with matching text. You can search for the next or previous commit with m
...
+Tabs
+----
+
+:ref:`settings-tabs`
+
+Commit
+^^^^^^
+
+Shows conformation or the currently selected commit.
+This tab can be moved to the revision grid.
```
Comment (RussKie) [inline source/browse_repository.rst:110]: Typos? I'm struggling with this...

File: source/browse_repository.rst [hunk 7]
```diff
@@ -35,26 +54,96 @@ commit with matching text. You can search for the next or previous commit with m
...
+
+Optional, only shown if the build server plugin is configured for the repo and if it has found a build for the selected commit.
+:ref:`settings-tabs-show-build-result-page`
+
+.. _browse-side-panel:
+
+Sidepanel
+---------
+
+Show submodules, local and remote branches, tags and remotes.
```
Comment (RussKie) [inline source/browse_repository.rst:149]:
```suggestion
Show local and remote branches, git remotes, tags and submodules.
```

**Review by RussKie — COMMENTED** [review]
👍

**Comment by RussKie** [comment]
Thank you @gerhardol  for doing this

File: source/browse_repository.rst [hunk 8]
```diff
@@ -4,10 +4,31 @@ Browse Repository
...
 
-View revision graph
--------------------
+TBD This page is incomplete
+
+The Browse window contains of several parts:
+
+* `Main toolbar`_
+* `Revision graph`_
+* `Tabs`_
```
[the remark this answers is not in the document]
Comment (gerhardol) [reply source/browse_repository.rst:13]: I find `Tabs` better describing the area

File: source/browse_repository.rst (hunk 5 above)
Comment (gerhardol) [reply source/browse_repository.rst:99]: ok

File: source/browse_repository.rst (hunk 6 above)
Comment (gerhardol) [reply source/browse_repository.rst:110]: `Commit information and message as well as branch and tag information for the selected commit.`

Especially this page in the doc should be improved. This PR is hopefully enough improvements to not be embarrassing.
I may make another stab at it some time, but no short term plans.

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Merge?

File: source/settings.rst (hunk 2 above)
Comment (gerhardol) [reply source/settings.rst]: Updating all

**Pull request merged**